### ISSUE TITLE
[TAMA][AOSP11] Google Camera doesn't work with latest OEMv5

### ISSUE THREAD

**Issue by jevgenp** [sonyxperiadev/bug_tracker#695]
**Platform:** TAMA
**Device:** XZ2 Compact
**Kernel version:** 4.14
**Android version:** 11 r36 (from actual sources at this moment)
**Software binaries version:** OEM v5

**Previously working on**
None (Worked only on A9)

**Description**
Google Cameras crashes on start (list of tried versions on screenshot), except that one worked on AOSP A9 (6.2.030).
This version starts, but viewfinder freezes (helps enabling of buffer fix in settings), and doesn't react on a capture button.

**Symptoms**
Crashes on start/Doesn't react on a capture button.

**How to reproduce**
Install any of versions from screenshot.

**Additional context**
![119940683-39c3ce80-bf98-11eb-9d90-3ed0e216ed1b](https://user-images.githubusercontent.com/66970854/119964290-bca55300-bfb1-11eb-8a0e-f9e0556c72e4.png)
![119941085-ca9aaa00-bf98-11eb-8964-a7f078a38154](https://user-images.githubusercontent.com/66970854/119964346-ca5ad880-bfb1-11eb-81b3-295112056ac8.png)
[gcam.txt](https://github.com/sonyxperiadev/bug_tracker/files/6559993/gcam.txt)

**Comment by MartinX3** [comment]
After the fix in the new OEMv5 we catched the next bug on our way to gcam happiness:
`java.lang.IllegalArgumentException: supportsCameraApi:2340: Unknown camera ID 2`

**Comment by sraase** [comment]
Akari only has two sensors, so only cameraId 0 and 1 are valid.
I don't know why the apps crash trying to access cameraId 2.
Can you post the output of "adb shell -n dumpsys media.camera"?

**Comment by jevgenp** [comment]
Here it is
[log.txt](https://github.com/sonyxperiadev/bug_tracker/files/6560211/log.txt)

**Comment by sraase** [comment]
Thank you.
Both camera HAL and MediaProfiles only list two cameraIds, which is correct.
I don't know why the apps try to open an additional camera. For now, I call "app bug", but that is probably too short.

**Comment by jevgenp** [comment]
Ok. I'll try to find some info how to change these id's in decompiled app. Thanks anyway.

**Comment by MartinX3** [comment]
Could it be possible, that the google camera starts counting the camera IDs at 1, while at SODP we start at 0?

**Comment by jevgenp** [comment]
@MartinX3 No idea, I'm just regular user.

**Comment by jevgenp** [comment]
I tried few more apps that uses raw for capturing and processing (PhotonCam, GelioSoft Camera2), but they also crashes.
But raw capture over SnapCamera HDR is working fine.
[3rd-apps.txt](https://github.com/sonyxperiadev/bug_tracker/files/6564821/3rd-apps.txt)
Don't know is it related.

**Comment by MartinX3** [comment]
@sraase Do you think that problem is in the OEM part or in the kernel? :)

TL;DR:
```
05-29 17:28:28.146   703  2287 E CHIUSECASE: [ERROR  ] chxextensionmodule.cpp:521 ExtendOpen() ExtendOpen failed! HW resource insufficient! openCameraCost=50CostOfAnyCurrentlyOpenLogicalCameras =100, m_totalResourceBudget = 100
```

```
11 E Camera3-Device: Camera 0: initialize: Could not open camera session: Function not implemented (-38)
```

Full Log Section:
```
05-29 17:28:28.117   819  7811 I CameraService: CameraService::connect call (PID -1 "com.geliosoft.geliosoftcamera2", camera ID 0) for HAL version default and Camera API version 2
05-29 17:28:28.123 17088 19744 D CCodecBufferChannel: [c2.android.vorbis.decoder#541] MediaCodec discarded an unknown buffer
05-29 17:28:28.124   819  7811 I Camera2ClientBase: Camera 0: Opened. Client: com.geliosoft.geliosoftcamera2 (PID 17088, UID 10102)
05-29 17:28:28.125 17088 19744 D CCodecBufferChannel: [c2.android.vorbis.decoder#541] MediaCodec discarded an unknown buffer
05-29 17:28:28.125   819  7811 I CameraDeviceClient: CameraDeviceClient 0: Opened
05-29 17:28:28.129 17088 19756 I hw-BpHwBinder: onLastStrongRef automatically unlinking death recipients
05-29 17:28:28.129   869  2239 I hw-BpHwBinder: onLastStrongRef automatically unlinking death recipients
05-29 17:28:28.132   869  2239 V C2Store : in ~ComponentModule
05-29 17:28:28.133   869  2239 V C2Store : unloading dll
05-29 17:28:28.143   819  7811 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
05-29 17:28:28.144   819  7811 I CameraProviderManager: Camera device device@3.5/legacy/0 torch status is now NOT_AVAILABLE
05-29 17:28:28.144   819  7811 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=0
05-29 17:28:28.144  1543  1585 D audio_hw_primary: adev_set_parameters: enter: cameraFacing=back
05-29 17:28:28.145   819  7811 I CameraProviderManager: Camera device device@3.5/legacy/1 torch status is now NOT_AVAILABLE
05-29 17:28:28.145   819  7811 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=1, newStatus=0
05-29 17:28:28.145   819  7811 E CameraService: onTorchStatusChangedLocked: cannot get torch status of camera 1: No such file or directory (-2)
05-29 17:28:28.146   703  2287 E CHIUSECASE: [ERROR  ] chxextensionmodule.cpp:521 ExtendOpen() ExtendOpen failed! HW resource insufficient! openCameraCost=50CostOfAnyCurrentlyOpenLogicalCameras =100, m_totalResourceBudget = 100
05-29 17:28:28.146   819  7811 I CameraProviderManager: Camera device device@3.5/legacy/0 torch status is now AVAILABLE_OFF
05-29 17:28:28.146   819  7811 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
05-29 17:28:28.147   819  7811 I CameraProviderManager: Camera device device@3.5/legacy/1 torch status is now AVAILABLE_OFF
05-29 17:28:28.147   819  7811 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=1, newStatus=1
05-29 17:28:28.148   819  7811 E CameraService: onTorchStatusChangedLocked: cannot get torch status of camera 1: No such file or directory (-2)
05-29 17:28:28.148   703  2287 E CamDev@3.2-impl: open: cannot open camera 0!
05-29 17:28:28.148   819  7811 E Camera3-Device: Camera 0: initialize: Could not open camera session: Function not implemented (-38)
05-29 17:28:28.149   819  7811 E Camera2ClientBase: initializeImpl: Camera 0: unable to initialize device: Function not implemented (-38)
05-29 17:28:28.149   819  7811 E CameraService: connectHelper: Could not initialize client from HAL.
05-29 17:28:28.149   819  7811 I Camera3-Device: disconnectImpl: E
05-29 17:28:28.151   819  7811 I CameraService: disconnect: Disconnected client for camera 0 for PID 17088
05-29 17:28:28.151   819  7811 I Camera2ClientBase: Closed Camera 0. Client was: com.geliosoft.geliosoftcamera2 (PID 17088, UID 10102)
05-29 17:28:28.153   819  7811 I Camera3-Device: disconnectImpl: E
```

**Comment by jevgenp** [comment]
@MartinX3 I think after one camera crash camera was unaccessable for another, i'll upload now correct ones.

**Comment by MartinX3** [comment]
```
05-29 17:53:32.546   812  1436 D CameraTraces: Process trace saved. Use dumpsys media.camera to view.
05-29 17:53:32.546   812  1436 E CameraDeviceClient: endConfigure: Camera 0: Error configuring streams: Function not implemented (-38)
05-29 17:53:32.547  6389  6599 E CameraCaptureSession: Session 1: Failed to create capture session; configuration failed
05-29 17:53:32.548  6389  6418 W vs.photoncamer: Long monitor contention with owner CameraBackground (6599) at void android.hardware.camera2.impl.CameraDeviceImpl.waitUntilIdle()(CameraDeviceImpl.java:1275) waiters=0 in void android.hardware.camera2.impl.CameraDeviceImpl$CameraDeviceCallbacks.onResultReceived(android.hardware.camera2.impl.CameraMetadataNative, android.hardware.camera2.impl.CaptureResultExtras, android.hardware.camera2.impl.PhysicalCaptureResultInfo[]) for 717ms
05-29 17:53:32.549  3714  3714 V ClearcutMetricXmitter: Transmission is done.
05-29 17:53:32.550  6389  6599 W System.err: android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): endConfigure:510: Camera 0: Error configuring streams: Function not implemented (-38)
05-29 17:53:32.551  6389  6599 W System.err: 	at android.hardware.camera2.CameraManager.throwAsPublicException(CameraManager.java:1009)
05-29 17:53:32.551  6389  6599 W System.err: 	at android.hardware.camera2.impl.ICameraDeviceUserWrapper.endConfigure(ICameraDeviceUserWrapper.java:119)
05-29 17:53:32.551  6389  6599 W System.err: 	at android.hardware.camera2.impl.CameraDeviceImpl.configureStreamsChecked(CameraDeviceImpl.java:484)
05-29 17:53:32.551  6389  6599 W System.err: 	at android.hardware.camera2.impl.CameraDeviceImpl.createCaptureSessionInternal(CameraDeviceImpl.java:679)
05-29 17:53:32.552  6389  6599 W System.err: 	at android.hardware.camera2.impl.CameraDeviceImpl.createCaptureSession(CameraDeviceImpl.java:527)
05-29 17:53:32.552  6389  6599 W System.err: 	at com.particlesdevs.photoncamera.capture.CaptureController.createCameraPreviewSession(CaptureController.java:1092)
05-29 17:53:32.553  6389  6599 W System.err: 	at com.particlesdevs.photoncamera.capture.CaptureController.captureStillPicture(CaptureController.java:1347)
05-29 17:53:32.553  6389  6599 W System.err: 	at com.particlesdevs.photoncamera.capture.CaptureController.access$300(CaptureController.java:118)
05-29 17:53:32.553  6389  6599 W System.err: 	at com.particlesdevs.photoncamera.capture.CaptureController$3.process(CaptureController.java:349)
05-29 17:53:32.553  6389  6599 W System.err: 	at com.particlesdevs.photoncamera.capture.CaptureController$3.onCaptureProgressed(CaptureController.java:394)
05-29 17:53:32.553  6389  6599 W System.err: 	at android.hardware.camera2.impl.CameraCaptureSessionImpl$1.lambda$onCaptureProgressed$2$CameraCaptureSessionImpl$1(CameraCaptureSessionImpl.java:678)
05-29 17:53:32.553  6920  6920 D ApplicationLoaders: Returning zygote-cached class loader: /system/framework/android.test.base.jar
05-29 17:53:32.554  6389  6599 W System.err: 	at android.hardware.camera2.impl.-$$Lambda$CameraCaptureSessionImpl$1$7mSdNTTAoYA0D3ITDxzDJKGykz0.run(Unknown Source:8)
05-29 17:53:32.554  6389  6599 W System.err: 	at android.os.Handler.handleCallback(Handler.java:938)
05-29 17:53:32.554  6389  6599 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:99)
05-29 17:53:32.554  6389  6599 W System.err: 	at android.os.Looper.loop(Looper.java:223)
05-29 17:53:32.555  6389  6599 W System.err: 	at android.os.HandlerThread.run(HandlerThread.java:67)
05-29 17:53:32.555  6389  6599 W System.err: Caused by: android.os.ServiceSpecificException: endConfigure:510: Camera 0: Error configuring streams: Function not implemented (-38) (code 10)
05-29 17:53:32.555  6389  6599 W System.err: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2387)
05-29 17:53:32.555  6389  6599 W System.err: 	at android.os.Parcel.createException(Parcel.java:2357)
05-29 17:53:32.556  6389  6599 W System.err: 	at android.os.Parcel.readException(Parcel.java:2340)
05-29 17:53:32.556  6389  6599 W System.err: 	at android.os.Parcel.readException(Parcel.java:2282)
05-29 17:53:32.556  6389  6599 W System.err: 	at android.hardware.camera2.ICameraDeviceUser$Stub$Proxy.endConfigure(ICameraDeviceUser.java:799)
05-29 17:53:32.556  6389  6599 W System.err: 	at android.hardware.camera2.impl.ICameraDeviceUserWrapper.endConfigure(ICameraDeviceUserWrapper.java:116)
05-29 17:53:32.556  6389  6599 W System.err: 	... 14 more
05-29 17:53:32.559  1070  2165 I chatty  : uid=1000(system) Binder:1070_9 expire 10 lines
05-29 17:53:32.570     0     0 E binder_alloc: 3461: binder_alloc_buf size 1056768 failed, no address space
05-29 17:53:32.570     0     0 E binder_alloc: allocated: 584 (num: 8 largest: 192), free: 1039800 (num: 4 largest: 1038272)
05-29 17:53:32.571     0     0 I binder  : 6198:6331 transaction failed 29201/-28, size 1056768-0 line 3207
05-29 17:53:32.574   812  1436 I Camera3-Device: disconnectImpl: E
05-29 17:53:32.574   812  1436 E Camera3-Device: Camera 0: disconnectImpl: Shutting down in an error state
05-29 17:53:32.574   812  1436 I CameraLatencyHistogram: ProcessCaptureRequest latency histogram (48) samples:
05-29 17:53:32.574   812  1436 I CameraLatencyHistogram:        40     80    120    160    200    240    280    320    360    inf (max ms)
05-29 17:53:32.575   812  1436 I CameraLatencyHistogram:      97.92   0.00   2.08   0.00   0.00   0.00   0.00   0.00   0.00   0.00 (%)
05-29 17:53:32.575   812  1436 I CameraProviderManager: Camera device device@3.5/legacy/0 torch status is now AVAILABLE_OFF
05-29 17:53:32.575   812  1436 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=0, newStatus=1
05-29 17:53:32.576   812  1436 I CameraProviderManager: Camera device device@3.5/legacy/1 torch status is now AVAILABLE_OFF
05-29 17:53:32.576   812  1436 I CameraService: onTorchStatusChangedLocked: Torch status changed for cameraId=1, newStatus=1
05-29 17:53:32.577  6920  6920 W le.android.kee: JIT profile information will not be recorded: profile file does not exist.
05-29 17:53:32.577   812  1436 E CameraService: onTorchStatusChangedLocked: cannot get torch status of camera 1: No such file or directory (-2)
05-29 17:53:32.577  6920  6920 W le.android.kee: JIT profile information will not be recorded: profile file does not exist.
05-29 17:53:32.578  6920  6920 W le.android.kee: JIT profile information will not be recorded: profile file does not exist.
05-29 17:53:32.582  6920  6920 D NetworkSecurityConfig: No Network Security Config specified, using platform default
05-29 17:53:32.583  6920  6920 D NetworkSecurityConfig: No Network Security Config specified, using platform default
05-29 17:53:32.598  6145  6168 I .android.diale: Waiting for a blocking GC ProfileSaver
05-29 17:53:32.605  6198  6335 I chatty  : uid=10167(com.google.android.gms) expire 22 lines
05-29 17:53:32.611     0     0 I CAM_INFO: CAM-SENSOR: cam_sensor_driver_cmd: 858 CAM_RELEASE_DEV Success, sensor_id:0x400,sensor_slave_addr:0x20
05-29 17:53:32.612     0     0 I CAM_INFO: CAM-SENSOR: msm_cam_sensor_handle_reg_gpio: 1466 Input Parameters are not proper
05-29 17:53:32.614     0     0 I CAM_INFO: CAM-SENSOR: cam_sensor_util_request_gpio_table: 792 No GPIO data
05-29 17:53:32.618   812  1436 I Camera3-Device: disconnectImpl: X
05-29 17:53:32.619   812  1436 I CameraService: disconnect: Disconnected client for camera 0 for PID 6389
05-29 17:53:32.619  6389  6599 E CameraEventsListener: onFatalError: CameraDevice.StateCallback : onError() : cameraDevice = [android.hardware.camera2.impl.CameraDeviceImpl@88078a9], error = [4]
```

**Comment by MartinX3** [comment]
@jevgenp
Please do a restart, let photon crash and run `adb shell -n dumpsys media.camera` for us. :)

**Comment by jevgenp** [comment]
@MartinX3 
[dumpsys.txt](https://github.com/sonyxperiadev/bug_tracker/files/6564907/dumpsys.txt)

**Comment by sraase** [comment]
> 05-29 17:28:28.146   703  2287 E CHIUSECASE: [ERROR  ] chxextensionmodule.cpp:521 ExtendOpen() ExtendOpen failed! HW resource insufficient! openCameraCost=50CostOfAnyCurrentlyOpenLogicalCameras =100, m_totalResourceBudget = 100

This usually happens after an error when switching between apps. Many apps do not handle this well and forget to close the camera device, blocking it for other apps even when sent to the background. Please make sure to always stop camera apps completely before starting a new one.

The logs above seem truncated. Try to run "adb logcat -c -G128m" before creating the log. To check platform behaviour, we need the CamX/CHI output.

**Comment by MartinX3** [comment]
@sraase am I allowed to share the CamX/CHI debugging instructions you gave me via email? :)

**Comment by sraase** [comment]
To enable CamX/CHI logging, set
> systemLogEnable=TRUE
in /vendor/etc/camera/camxoverridesettings.txt and reboot.

In the same file, you can set logWarningMask, logInfoMask and logVerboseMask with a bitmask to enable additional logging (also requires a reboot). Don't overdo it, you can easily overload logcat and any human trying to read those.

**Comment by sraase** [comment]
Better not.

**Comment by MartinX3** [comment]
Okidoki, then I look the next time I can flash SODP on my phone and create the logs. :)
(At the moment busy with job and studies like always xD)

**Comment by jevgenp** [comment]
If still needed (don't know is that or not):
[geliosoft.txt](https://github.com/sonyxperiadev/bug_tracker/files/6570083/geliosoft.txt)
[photon.txt](https://github.com/sonyxperiadev/bug_tracker/files/6570087/photon.txt)
[gcam_8_2_viewfinder_freeze.txt](https://github.com/sonyxperiadev/bug_tracker/files/6570088/gcam_8_2_viewfinder_freeze.txt)
[gcam_6_2_30_no_reaction.txt](https://github.com/sonyxperiadev/bug_tracker/files/6570090/gcam_6_2_30_no_reaction.txt)

**Comment by MartinX3** [comment]
Dear @sraase here is the special log
[camera.log](https://github.com/sonyxperiadev/bug_tracker/files/6641968/camera.log)

I tested it with OpenGapps Pico and this [GCam](https://1-dontsharethislink.celsoazevedo.com/file/filesc/MGC_8.1.101_A9_GV1c_MGC.apk)

**Comment by sraase** [comment]
I don't get this application...

> 06-12 11:31:53.030  9094  9094 E ALL_TEST [ID = 0]: PASSED
> 06-12 11:31:53.034  9094  9094 E ALL_TEST [ID = 1]: PASSED
> 06-12 11:31:53.035  8975  8982 E CameraService: supportsCameraApi: Unknown camera ID 2
> 06-12 11:31:53.036  9094  9094 E LENS [ID = 2]: NOT_EXIST
> [... trying to search for other devices by probing, which is bad...]
> 06-12 11:31:53.161  9094  9094 E FOUND_LENSES:  [ 0 1 ]
> 06-12 11:31:53.162  9094  9094 E SORTS_LENSES:  [ 0 1 ]
> 06-12 11:31:53.162  9094  9094 E AFTER_HIDING_LENSES:  [ 1 ]
> [... found back and front cameras, but rejects the back camera, which is weird ...]
> 06-12 11:31:54.007  9094  9094 E AndroidRuntime: FATAL EXCEPTION: main
> 06-12 11:31:54.007  9094  9094 E AndroidRuntime: Process: com.android.MGC, PID: 9094
> 06-12 11:31:54.007  9094  9094 E AndroidRuntime: java.lang.IllegalArgumentException: supportsCameraApi:2365: Unknown camera ID 2
> [... then fails trying to open the camera which it already knows doesn't exist ...]

... it probes for all cameras (bad style, you can ask the system), then ignored the main camera (don't know any reason from the log), and then it fails trying to get the characteristics of a camera which it has already proven does not exist.

**Comment by jevgenp** [comment]
So it's google camera problem then? And any app that using raw?
If yes, why version that worked on AOSP9 doesn't work on AOSP11?

**Comment by MartinX3** [comment]
@sraase That's weird, I enabled the entire logging stuff.
And @jevgenp 
I try to get a gcam version without this weird camera id logic hack after speaking with some gcam developers.

Probably this will just work or we will be redirected to another error. I hope for the first one. :)

**Comment by jevgenp** [comment]
@MartinX3 I was talking about version that worked on A9 - Arnova's 1.6/2.2 (or San1ty 9.4/9.5).

**Comment by MartinX3** [comment]
@jevgenp according to their readme

    GCam 6.2 for Android 9 or newer. See GCam 6.1 below for Android 8.
    On some phones, GCam 6.x doesn't work on Android 10 or newer. Use GCam 7+.

**Comment by jevgenp** [comment]
@MartinX3 I know, but in this case the same version that worked on A9 (6.2.030) runs on A11, but when you press Capture button - processing message appears in notification area and doesn't disappear until you exit application. 
Maybe there should be specific package name of google camera to use hdr+?

**Comment by MartinX3** [comment]
@jevgenp Ah I thought you get the same crash on starting the app.
Then this app doesn't have this weird camera ID logic implemented.
I'll search this app and try it with logs or you give me the link if I don't find it? :)

**Comment by jevgenp** [comment]
@MartinX3 https://www.celsoazevedo.com/files/android/google-camera/f/changelog593/
Then go to Settings > About > Enable Advanced Mode and then HDR+ Control > find and enable Buffer fix

**Comment by MartinX3** [comment]
@sraase I was able to gather a log and receive a loop of errors.

[camera.log](https://github.com/sonyxperiadev/bug_tracker/files/6643925/camera.log)

```06-13 15:09:52.750 10093 10116 I CHISENSORSONY: IMX400: set gain 3.07 (analog 3.07 digital 1.00 isp 1.00 lineCount 11602)
06-13 15:09:52.758 10093 10115 E CamX    : [ERROR][FD     ] camxfdmanagernode.cpp:3598 PopulateGravityData() NCS Gravity handle is NULL
06-13 15:09:52.758 10736 10846 W CameraMetadataJV: Expect face scores and rectangles to be non-null
06-13 15:09:52.758 10736 10868 E OisListener: Null pointer for OIS data. OIS API version: 2
06-13 15:09:52.759 10736 10846 D CAM_RepeatingFRP: Processing fallback request
06-13 15:09:52.777 10093 10118 I CHIUSECASE: [REQMAP ] chxadvancedcamerausecase.cpp:6934 LogFeatureRequestMappings() frame: 357  <==>  chiOverrideFrameNum: 344  <==>  chiFrameNum: 344 -- RAW Jpeg Request
06-13 15:09:52.778 10093 10112 I CamX    : [REQMAP][CORE   ] camxsession.cpp:2744 ProcessRequest() chiFrameNum: 344  <==>  requestId: 345  <==>  sequenceId: 344  <==> CSLSyncId: 345  -- ZSLPreviewRaw_0
06-13 15:09:52.791 10093 10113 E CamX    : [ERROR][IQMod  ] camxipeica.cpp:1471 UpdatePerspectiveParamsToContext() Invalid warp data for path 0
06-13 15:09:52.792 10093 10111 I CHISENSORSONY: IMX400: set gain 3.07 (analog 3.07 digital 1.00 isp 1.00 lineCount 11602)```

Or this

```
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: Retrieving snapshot for com.google.android.apps.camera#org.codeaurora.snapcam failed
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: java.util.concurrent.TimeoutException: Timed out waiting for Task
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at lka.a(Unknown Source:58)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at ljg.a(Unknown Source:45)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at duv.run(Unknown Source:49)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at lqd.run(Unknown Source:2)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at java.lang.Thread.run(Thread.java:923)
06-13 15:08:24.456 10172 10211 E PhenotypeFlagCommitter: 	at loy.run(Unknown Source:5)
06-13 15:08:24.665 10172 10318 D CAM_WearMessageUtil: findBestNode failed!
```

**Comment by sraase** [comment]
Are you using a stock Open Devices build, or did you change it in any way?

**Comment by MartinX3** [comment]
I just compiled AOSP 11 and use OEMv5

**Comment by sraase** [comment]
I built AOSP 11 using OEMv5 on Xperia XZ2, installed the APK you linked to and received the following error:
> 06-14 11:56:11.560  3962  3962 E AndroidRuntime: FATAL EXCEPTION: main
> 06-14 11:56:11.560  3962  3962 E AndroidRuntime: Process: org.codeaurora.snapcam, PID: 3962
> 06-14 11:56:11.560  3962  3962 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.google.android.apps.camera.legacy.app.app.CameraApp: java.lang.SecurityException: Failed to find provider com.google.android.gsf.gservices for user 0; expected to find a valid ContentProvider for this authority
> 06-14 11:56:11.560  3962  3962 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6724)

**Comment by MartinX3** [comment]
@sraase do you have GAPPS installed?
I use OpenGapps Pico (the smallest one)
https://sourceforge.net/projects/opengapps/files/arm64/test/20210130/open_gapps-arm64-11.0-pico-20210130-TEST.zip/download
flashed with TWRP
https://www.dhsfileserver.de/ftp/MartinX3/Xperia%20XZ2/android_11/twrp/sodp/2021-03-02_17-06-00_twrp_akari.tar.gz

GAPPS are needing a fullwipe, otherwise problems occur.

- `fastboot reboot bootloader` or `adb reboot bootloader`
- `fastboot boot twrp.img`

Sometimes if I use the hardware buttons to enter the bootloader, I just get a black screen using `fastboot boot twrp.img`
Probably a bootloader bug?

**Comment by sraase** [comment]
Does the same issue happen with the front camera? If yes, then it is not a driver issue.

**Comment by jevgenp** [comment]
I'm far from programming, but you want to say that all of these apps that working with raw on a bunch of devices without problems doesn't work on sony and it is an apps issue? Sounds strange.

**Comment by sraase** [comment]
No, this is not what I want to say. The drivers are one part of the camera system. If a problem happens for both the front and back cameras, then the drivers are unlikely to be at fault (different drivers).

Thanks for the APK. Issue looks similar to #544 but caused by the app requesting RAW16 format.

**Comment by jevgenp** [comment]
Maybe some more info needed? As I see Martin didn't answer your question about front camera :)
I'm ready to flash again AOSP to collect any log.

And few questions off topic but related to camera: 
Is ISO 100 is a minimum? according to datasheet our module supports ISO 50.
And what about vignetting and manual exposure?

**Comment by MartinX3** [comment]
I just need some free time because of job, studies and a dataloss (ssd damage, but i have an older backup and was able to recover data from the failed ssd)

On the front camera the gcam did freeze too.

And i need to analyse why i can't record and send videos in telegram anynore since I use SODP (wrong video codec it says), but that's stuff for a new ticket.

**Comment by sraase** [comment]
> I'm ready to flash again AOSP to collect any log.
Currently not needed. I was able to reproduce the problem, but did not look into it yet.

> Is ISO 100 a minimum?
I suspect that the ISO values shown are simply wrong.

> And what about vignetting and manual exposure?
Please clarify your question.

**Comment by jevgenp** [comment]
> The ISO values shown are very likely wrong.
> 
> >

Ah ok, but in this case calculated exposure value is correct?
https://drive.google.com/drive/folders/1AAjCIUx6SnhtRFPQu19BQlKE4t5rT2eS?usp=sharing
Same app settings: Stock ISO40 @ 1/898, AOSP ISO100 @ 1/2705.
Maybe it's subjective but jpeg from AOSP more noisy for me (or it caused by underexposure)

> Please clarify your question.

Manual exposure, i.e. in SnapCamera HDR when I try to move exposure slider then this happens
![Screenshot_20210622-110240](https://user-images.githubusercontent.com/66970854/122893563-1b5eb200-d34f-11eb-8709-73b9ce0fed40.png)
As I remember on AOSP9 manual exposure worked, but minimal value was only 1/9.

What about vignetting - darkening of image corners when compared to the center.
https://drive.google.com/drive/folders/1hq3kKHhKvV0DJsZeKHtoIWrFCQMwa9zg?usp=sharing
![Screenshot_4](https://user-images.githubusercontent.com/66970854/122887880-f9aefc00-d349-11eb-8670-eed6bad5d81e.jpg)
![Screenshot_3](https://user-images.githubusercontent.com/66970854/122888059-28c56d80-d34a-11eb-95c9-e7ad5b52b847.jpg)
These images from google camera on AOSP9.
As I understand phone should store info about lens and apps can use it to correct this thing.
Or it can be corrected in RAW?

**Comment by sraase** [comment]
The actual exposure time (lineCount) and ISO (gain values) are shown in logcat.
> CHISENSORSONY: IMX400: set gain 2.03 (analog 2.03 digital 1.00 isp 1.00 lineCount 11602)

Camera output on AOSP will have increased noise compared to Xperia, especially in darkness. This is known and cannot be fixed.

About the vignetting, please remove /odm/lib64/camera/com.sony.eeprom.blakiston.so (this will disable LSC) and see if the situation improves.

**Comment by jevgenp** [comment]
> About the vignetting, please remove /odm/lib64/camera/com.sony.eeprom.blakiston.so (this will disable LSC) and see if the situation improves

Seems yes, thanks!

> Camera output on AOSP will have increased noise compared to Xperia, especially in darkness. This is known and cannot be fixed.

Noted.

And what about manual SS control?

**Comment by jevgenp** [comment]
> About the vignetting, please remove /odm/lib64/camera/com.sony.eeprom.blakiston.so (this will disable LSC) and see if the situation improves.



> Seems yes, thanks!
> 
> >

To early I made conclusion...
![Screenshot_9](https://user-images.githubusercontent.com/66970854/122950652-4c59d980-d385-11eb-9360-50136f143a6f.jpg)
![Screenshot_12](https://user-images.githubusercontent.com/66970854/122951095-b1153400-d385-11eb-876e-2910b8209650.jpg)
A little better but it still there (1st raw, 2nd jpeg).

**Comment by jevgenp** [comment]
Just tried august release with OEMv6, and now I'm unable to take raw even with SnapCamera HDR (that worked with previous build and OEMv5).
[raw.txt](https://github.com/sonyxperiadev/bug_tracker/files/7000369/raw.txt)

**Comment by sraase** [comment]
Not sure what's going on. The HAL appears to open the camera correctly, then Gcam (APK linked above) throws `java.lang.IllegalStateException: Unable to access OneCamera.` a few times and hangs on any Photo mode; Video seems fine. Can you confirm?

**Comment by jevgenp** [comment]
Are you talking about 6.2.030 version?

**Comment by sraase** [comment]
Gcam_6.2.030_Advanced_V2.2.190716.1800-Final.apk

**Comment by jevgenp** [comment]
@sraase same behavior. Only video mode working. In other modes just black screen for a while and app quits.
On previous build at least viewfinder worked :)
And as I said before SnapCamera unable to take raw anymore using c2api.
[gcam.txt](https://github.com/sonyxperiadev/bug_tracker/files/7015679/gcam.txt)

**Comment by mikkorantalainen** [comment]
Vignetting is caused by actual lens optics so RAW should always show it. Software can remove/hide it, of course, so JPEG may not show any vignetting.

**Comment by jevgenp** [comment]
Do you need any new log from latest r43+oemv7? Because situation didn't change.

**Comment by jerpelea** [comment]
Discontinued Android version
